clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_utils.py

```diff
@@ -0,0 +1,221 @@
+# coding: utf8
+
+def extract_slices(preprocessed_T1, slice_direction=0, slice_mode='original'):
+    """
+    This function extracts the slices from three directions
+    :param preprocessed_T1:
+    :param slice_direction: which axis direction that the slices were extracted
+    :return:
+    """
+    import torch
+    import os
+
+    image_tensor = torch.load(preprocessed_T1)
+    # reshape the tensor, delete the first dimension for slice-level
+    image_tensor = image_tensor.view(image_tensor.shape[1], image_tensor.shape[2], image_tensor.shape[3])
+
+    # sagital
+    slice_list_sag = range(20, image_tensor.shape[0] - 20)  # delete the first 20 slice and last 20 slices
+
+    basedir = os.getcwd()
+    output_file_original = []
+    output_file_rgb = []
+    if slice_direction == 0:
+        for index_slice, index_slice_list in zip(slice_list_sag, range(len(slice_list_sag))):
+            # for i in slice_list:
+            # sagital
+            slice_select_sag = image_tensor[index_slice, :, :]
+
+            # convert the slices to images based on if transfer learning or not
+            # train from scratch
+            extracted_slice_original_sag = slice_select_sag.unsqueeze(0)  # shape should be 1 * W * L
+
+            # train for transfer learning, creating the fake RGB image.
+            slice_select_sag = (slice_select_sag - slice_select_sag.min()) / (slice_select_sag.max() - slice_select_sag.min())
+            extracted_slice_rgb_sag = torch.stack((slice_select_sag, slice_select_sag, slice_select_sag))  # shape should be 3 * W * L
+
+            # save into .pt format
+            if slice_mode == 'original':
+                output_file_original.append(
+                        os.path.join(
+                            basedir,
+                            os.path.basename(preprocessed_T1).split('.pt')[0]
+                            + '_axis-sag_originalslice-'
+                            + str(index_slice)
+                            + '.pt'
+                            )
+                        )
+                torch.save(extracted_slice_original_sag.clone(), output_file_original[index_slice_list])
+            elif slice_mode == 'rgb':
+                output_file_rgb.append(
+                        os.path.join(
+                            basedir,
+                            os.path.basename(preprocessed_T1).split('.pt')[0]
+                            + '_axis-sag_rgbslice-'
+                            + str(index_slice)
+                            + '.pt'
+                            )
+                        )
+                torch.save(extracted_slice_rgb_sag.clone(), output_file_rgb[index_slice_list])
+
+    elif slice_direction == 1:
+        # cornal
+        slice_list_cor = range(15, image_tensor.shape[1] - 15)  # delete the first 20 slice and last 15 slices
```

Comment:
There is an inconsistence between what is done (15, 15) and what is commented (20, 15). This is also the case in other parts of this function.

Reply:
Fixed!